scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -1290,6 +1290,62 @@
             (catch Exception e
               (log/error e "Unable to decline offers!")))))))
 
+(defn create-mesos-scheduler
+  "Creates the mesos scheduler which offloads processing of status updates to an agent.
+   The agent is used to ensure in-order asynchronous processing of status updates."
```

Comment:
It is unbounded (limited by heap memory), no back pressure.

I thought about using multiple agents to get benefits of parallelization, but that requires parsing the message and using the status id to still ensure in-order processing of messages from a given task.